spring-ai-core/src/main/java/org/springframework/ai/chat/client/advisor/api/AdvisedRequest.java

```diff
@@ -57,13 +57,53 @@
  * @param advisorParams the map of advisor parameters
  * @param adviseContext the map of advise context
  * @param toolContext the tool context
+ * @author Christian Tzolov
+ * @author Thomas Vitale
+ * @since 1.0.0
  */
-public record AdvisedRequest(ChatModel chatModel, String userText, String systemText, ChatOptions chatOptions,
-		List<Media> media, List<String> functionNames, List<FunctionCallback> functionCallbacks, List<Message> messages,
-		Map<String, Object> userParams, Map<String, Object> systemParams, List<Advisor> advisors,
-		Map<String, Object> advisorParams, Map<String, Object> adviseContext, Map<String, Object> toolContext) {
+public record AdvisedRequest(
+// @formatter:off
+		ChatModel chatModel,
+		String userText,
```

Comment:
may be an odd case, but maybe should allow userText to be null?

Reply:
I thought a lot about a possible use case where I would want that and didn't really come up with anything. For example, OpenAI requires a message to be included in each request to a chat model. It fails if the user message is null. Before this change, we supported the user message to be null, and in that case set the "userText" to be an empty string as a workaround. That could be dangerous because as a developer I wouldn't catch scenarios where I'm not passing a user message by mistake, but the model would still be called and the outcome would be unpredictable.